src/Toolkit/PubSub/V1/PublisherApi.php

```diff
@@ -0,0 +1,671 @@
+            $this->stub,
+            'ListTopicSubscriptions',
+            $mergedSettings,
+            $this->descriptors['listTopicSubscriptions']
+        );
+
+        return $callable(
+            $request,
+            [],
+            ['call_credentials_callback' => $this->createCredentialsCallback()]);
+    }
+
+    /**
+     * Deletes the topic with the given name. Generates `NOT_FOUND` if the topic
+     * does not exist. After a topic is deleted, a new topic may be created with
+     * the same name; this is an entirely new topic with none of the old
+     * configuration or subscriptions. Existing subscriptions to this topic are
+     * not deleted, but their `topic` field is set to `_deleted-topic_`.
+     *
+     * Sample code:
+     * ```
+     * try {
+     *     $publisherApi = new PublisherApi();
+     *     $formattedTopic = PublisherApi::formatTopicName("[PROJECT]", "[TOPIC]");
+     *     $publisherApi->deleteTopic($formattedTopic);
+     * } finally {
+     *     if (isset($publisherApi)) {
+     *         $publisherApi->close();
+     *     }
+     * }
+     * ```
+     *
+     * @param string $topic        Name of the topic to delete.
+     * @param array  $optionalArgs {
+     *                             Optional. There are no optional parameters for this method yet;
+     *                             this $optionalArgs parameter reserves a spot for future ones.
+     *                             }
+     * @param array  $callSettings {
+     *                             Optional.
+     *
+     *    @var Google\GAX\RetrySettings $retrySettings
+     *          Retry settings to use for this call. If present, then
+     *          $timeout is ignored.
+     *    @var int $timeoutMillis
+     *          Timeout to use for this call. Only used if $retrySettings
+     *          is not set.
+     * }
+     *
+     * @throws Google\GAX\ApiException if the remote call fails
+     */
+    public function deleteTopic($topic, $optionalArgs = [], $callSettings = [])
+    {
+        $request = new DeleteTopicRequest();
+        $request->setTopic($topic);
+
+        $mergedSettings = $this->defaultCallSettings['deleteTopic']->merge(
+            new CallSettings($callSettings)
+        );
+        $callable = ApiCallable::createApiCall(
+            $this->stub,
+            'DeleteTopic',
+            $mergedSettings,
+            $this->descriptors['deleteTopic']
+        );
+
+        return $callable(
+            $request,
+            [],
+            ['call_credentials_callback' => $this->createCredentialsCallback()]);
+    }
+
+    /**
+     * Initiates an orderly shutdown in which preexisting calls continue but new
+     * calls are immediately cancelled.
+     */
+    public function close()
+    {
+        $this->stub->close();
+    }
+
+    private function createCredentialsCallback()
+    {
+        return $this->grpcBootstrap->createCallCredentialsCallback($this->scopes);
```

Comment:
@garrettjonesgoogle @jgeewax Just wanted to follow up on this one again. Any thoughts on allowing users to pass in a keyfile? This is something we support for the REST side of things and I imagine it would make sense to keep the same functionality for gRPC.